src/extension_commands.ml

```diff
@@ -766,6 +766,121 @@ module MerlinJump = struct
     command Extension_consts.Commands.merlin_jump handler
 end
 
+module Navigate_holes = struct
+  let extension_name = "Navigate between Typed Holes"
+
+  let ocaml_lsp_doesnt_support_typed_holes ocaml_lsp =
+    not (Ocaml_lsp.can_handle_typed_holes ocaml_lsp)
+
+  let is_valid_text_doc textdoc =
+    match TextDocument.languageId textdoc with
+    | "ocaml" | "ocaml.interface" | "reason" -> true
+    | _ -> false
+
+  let send_request_to_lsp client doc =
+    let uri = TextDocument.uri doc in
+    Custom_requests.send_request client Custom_requests.typedHoles uri
+
+  let display_results (results : Range.t list) text_document =
+    let quickPickItems =
+      List.map results ~f:(fun res ->
+          let line = Position.line @@ Range.end_ res in
+          ( QuickPickItem.create
+              ~label:
+                (Printf.sprintf
+                   "Line %d: %s"
+                   line
+                   (TextLine.text @@ TextDocument.lineAt ~line text_document))
+              ()
+          , (res, ()) ))
```

Comment:
Maybe we should use the `details`of the quick pick item to show the line ?

Also, I am wondering whether it would be better to navigate as soon as the user selects a quickpick (without validating with Enter) that would allows the user to see the context around the hole more easily before making a decision.

Reply:
Do you mean to pass the line number to the `~label` and the content to the `~detail`?

Reply:
It's passing the content of the line to `~detail`.
After https://github.com/ocamllabs/vscode-ocaml-platform/pull/1626 is merged, there are some additional modules that can allow us use the low level `QuickPick` component which will improve the functionality of navigating typed holes much better.

Reply:
Merged.

Reply:
Thanks, I'll do the necessary updates and then this PR will be ready for merging :)